base/client.jl

```diff
@@ -548,13 +538,28 @@ function _start()
     append!(ARGS, Core.ARGS)
     # clear any postoutput hooks that were saved in the sysimage
     empty!(Base.postoutput_hooks)
+    local ret = 0
     try
-        exec_options(JLOptions())
+        should_run_repl = exec_options(JLOptions())
+        if should_run_repl
+            if isassigned(REPL_MODULE_REF)
+                ret = REPL_MODULE_REF[].main(ARGS)
+            end
+        elseif isdefined(Main, :main)
+            if Core.Compiler.generating_sysimg()
+                precompile(Main.main, (typeof(ARGS),))
+            else
+                ret = invokelatest(Main.main, ARGS)
+            end
+        end
```

Comment:
I thought the triage discussion was to have it call `run_main_repl` if `-i` was passed or `Main.main` was undefined, and otherwise to run `Main.main`, and otherwise to exit

Reply:
So the concern is in particular if Main.main is defined, but `-i` is not specified, but also we don't have a program file (e.g. because we loaded a sysimage with defined Main.main)? Yeah, I agree that shouldn't run the REPL, I'll tweak it.

Reply:
yep